packages/shared/src/mock/metadata.ts

```diff
@@ -29,14 +30,26 @@ export function makeMockFile(): MockFile {
     children: [
       group('entities', [
         group('empty_group'),
-        dataset('empty_dataset', { class: DTypeClass.Unknown }, null, null),
+        dataset('empty_dataset', unknownType(), null, null),
         datatype('datatype', compoundType({ int: intType() })),
         scalar('raw', { int: 42 }),
         scalar('raw_large', undefined), // generated dynamically by `MockProvider`
-        scalar('scalar_num', 0),
-        scalar('scalar_str', 'foo'),
-        scalar('scalar_bool', true),
-        scalar('scalar_cplx', [1, 5]),
+        scalar('scalar_num', 0, { attributes: [scalarAttr('attr', 0)] }),
+        scalar('scalar_str', 'foo', {
+          attributes: [scalarAttr('attr', 'foo')],
+        }),
+        scalar('scalar_bool', true, { attributes: [scalarAttr('attr', true)] }),
+        scalar('scalar_cplx', [1, 5], {
+          attributes: [scalarAttr('attr', cplx(1, 5))],
+        }),
+        scalar('scalar_compound', ['foo', 2], {
+          type: compoundType({ str: strType('ASCII', 3), int: intType(8) }),
+          attributes: [
+            scalarAttr('attr', ['foo', 2], {
+              type: compoundType({ str: strType('UTF-8'), int: intType(8) }),
+            }),
+          ],
+        }),
```

Comment:
Adding a scalar compound dataset. It's currently rendered with the Raw visualization as ["foo",2], so the idea will be to improve on that at some point.